junifer/markers/collection.py

```diff
@@ -131,14 +131,19 @@ def validate(self, datagrabber: "BaseDataGrabber") -> None:
         logger.info(f"Data Reader output type: {t_data}")
 
         if self._preprocessors is not None:
+            validated_input_data_types = []
             for preprocessor in self._preprocessors:
                 logger.info(
                     "Validating Preprocessor: "
                     f"{preprocessor.__class__.__name__}"
                 )
                 # Validate preprocessor
-                t_data = preprocessor.validate(t_data)
-                logger.info(f"Preprocess output type: {t_data}")
+                p_data = preprocessor.validate(t_data)
+                logger.info(f"Preprocess output type: {p_data}")
+                # Extend validated list
+                validated_input_data_types.extend(p_data)
+            # Set t_data for further use
+            t_data = list(set(validated_input_data_types))
```

Comment:
This will not work in case you have a preprocessing step that only modifies one type and passes the others through.

Basically, after pre-processing you should have AT LEAST the same elements as before pre-processing:

`t_data = list(set(validated_input_data_types) | set(t_data))`

Reply:
Yeah agree, good point, made the change.